projects/kit/components/input-phone-international/input-phone-international.component.ts

```diff
@@ -151,15 +154,20 @@ export class TuiInputPhoneInternationalComponent
     @HostListener('paste.capture.prevent.stop', ['$event'])
     @HostListener('drop.capture.prevent.stop', ['$event'])
     onPaste(event: ClipboardEvent | DragEvent): void {
-        let value = extractValueFromEvent(event).replace(TUI_NON_DIGITS_REGEXP, '');
-        const countryIsoCode = this.extractCountryCode(value);
+        let value = tuiExtractValueFromEvent(event).replace(TUI_NON_DIGITS_REGEXP, '');
+        const countryIsoCode = this.extractCountryCodePipe.transform(
+            value,
+            this.countries,
+        );
 
         if (!countryIsoCode) {
-            this.updateValue(
-                `${this.inputPhoneCountryCode}${value}`
-                    .replace(TUI_MASK_SYMBOLS_REGEXP, '')
-                    .slice(0, this.getMaxAllowedLength(this.countryIsoCode)),
-            );
+            // @ts-ignore
+            this.value = `${this.inputPhoneCountryCode}${value}`
```

Comment:
`AbstractTuiControl` doesn't have this setter in `v2.x`.
In `2.x` you should use `updateValue(...)`.

___
Setter `value` of `AbstractTuiControl` was released only in 3.23.0:
* https://github.com/taiga-family/taiga-ui/pull/4048

Reply:
thanks, fixed it